pilot/pkg/proxy/envoy/v2/monitoring.go

```diff
@@ -158,9 +158,10 @@ var (
 
 func recordSendError(metric monitoring.Metric, err error) {
 	s, ok := status.FromError(err)
-	// Unavailable code will be sent when a connection is closing down. This is very normal,
+	// Unavailable or canceled code will be sent when a connection is closing down. This is very normal,
 	// due to the XDS connection being dropped every 30 minutes, or a pod shutting down.
-	if !ok || s.Code() != codes.Unavailable {
+	isError := s.Code() != codes.Unavailable && s.Code() != codes.Canceled
```

Comment:
why do you process these two code differently? They are also error code, and it seems `DeadlineExceeded` is also a temp failure.

Reply:
We have a grafana dashboard that shows "pilot errors". When a pod shuts down, it cancels the context, which results in a "pilot error" when there was no error. The context being canceled is not an error so we shouldn't report it as one

Reply:
I donot think all other codes indicate pilot error, at least `DeadlineExceeded` doesnot.
But i agree they belong to `SendError`.